fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/impl/resource/loader/ModResourcePackUtil.java

```diff
@@ -82,6 +92,36 @@ public static InputStream openDefault(ModMetadata info, ResourceType type, Strin
 		}
 	}
 
+	// Basically a copy of the vanilla method, with an additional parameter for Text
+	@Nullable
+	public static ResourcePackProfile of(String name, Text text, boolean alwaysEnabled, Supplier<ResourcePack> packFactory, ResourcePackProfile.Factory profileFactory, ResourcePackProfile.InsertionPosition insertionPosition, ResourcePackSource packSource) {
```

Comment:
It's still awful. If you run `genSources`, CFR would output nicer code than FF one.